src/XamlX/Ast/Intrinsics.cs

```diff
@@ -136,6 +136,78 @@ IXamlType ResolveReturnType()
         public IXamlAstTypeReference Type => new XamlAstClrTypeReference(this, ResolveReturnType(), false);
     }
 
+#if !XAMLX_INTERNAL
+    public
+#endif
+    class XamlArrayExtensionNode : XamlAstNode, IXamlAstValueNode, IXamlAstEmitableNode<IXamlILEmitter, XamlILNodeEmitResult>
+    {
+        private IXamlAstValueNode _elementType;
+        private XamlAstClrTypeReference _type;
+
+        public XamlArrayExtensionNode(
+            XamlAstObjectNode lineInfo,
+            IXamlAstValueNode elementType,
+            IList<IXamlAstValueNode> elements)
+            : base(lineInfo)
+        {
+            _elementType = elementType;
+
+            Elements = elements;
+        }
+
+        public IXamlAstTypeReference Type =>
+            _type ?? throw new XamlParseException("Could not resolve x:Array Type for an unknown reason! Try using x:Type to specify the array element type.", this);
+
+        public IList<IXamlAstValueNode> Elements { get; }
+
+        public override void VisitChildren(Visitor visitor)
+        {
+            _elementType = (IXamlAstValueNode)_elementType.Visit(visitor);
+
+            if (_elementType is XamlTypeExtensionNode n && n.Value is XamlAstClrTypeReference clr)
+            {
+                _type = new XamlAstClrTypeReference(this, clr.Type.MakeArrayType(1), false);
+            }
+
+            for (int i = 0; i < Elements.Count; i++)
+            {
+                Elements[i] = (IXamlAstValueNode)Elements[i].Visit(visitor);
+            }
+        }
+
+        public XamlILNodeEmitResult Emit(XamlEmitContext<IXamlILEmitter, XamlILNodeEmitResult> context, IXamlILEmitter codeGen)
```

Comment:
The current policy is to have a separate emitter class rather then implementing IXamlAstEmitableNode. That's needed because of the planned C++ emitter backend for UWP.